src/main/java/frc/robot/subsystems/LightStrip.java

```diff
@@ -1,49 +1,39 @@
 package frc.robot.subsystems;
 
 import edu.wpi.first.wpilibj.motorcontrol.Spark;
-import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.constants.Constants;
 import frc.robot.utils.BlinkinPattern;
-import frc.robot.utils.logging.Logger;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
 import java.util.function.BooleanSupplier;
 
 public class LightStrip extends SubsystemBase {
     private static final String baseLogName = "/robot/lightstrip/";
     private final Spark colorSensorPort;
-    private BlinkinPattern pattern = BlinkinPattern.WHITE;
+    private BlinkinPattern pattern = BlinkinPattern.BLACK;
     private final Map<BooleanSupplier, BlinkinPattern> predicateLightEvents = new HashMap<>();
-
     public LightStrip(int port) {
         this.colorSensorPort = new Spark(port);
-    }
-
-    @Override
-    public void periodic() {
-        predicateLightEvents.keySet()
+        Executors.newScheduledThreadPool(1).scheduleAtFixedRate(() -> predicateLightEvents.keySet()
                 .stream()
```

Comment:
This is likely wasteful in new classes and garbage collection. Also, instead of t a new executor we may rely on a simpler mechanism

Reply:
made into a thread